src/views/AddItem.jsx

```diff
@@ -1,3 +1,93 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { addItem } from '../api/firebase';
+import { useState } from 'react';
+
+export function AddItem({ listId }) {
+	// declare itemName, daysUntilNextPurchase, and radioSelect
+	// into separate state variables
+	const [itemName, setItemName] = useState('');
+
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(null);
+
+	const [radioSelect, setRadioSelect] = useState('soon');
+
+	// Store values inside itemData and sent over to database
+	// otherwise log an error if request fails
+	const onFormSubmit = (event) => {
+		event.preventDefault();
+		const itemData = {
+			itemName,
+			daysUntilNextPurchase,
+		};
+		try {
+			addItem(listId, itemData);
+			alert(`${itemData.itemName} was added to your list`);
+		} catch (error) {
+			console.log(error);
+		}
+	};
+
+	// Get user input and set itemName
+	const handleItemName = (event) => {
+		setItemName(event.target.value);
+	};
+
+	// Get radio selection and set daysUntilNextPurchase
+	const handlePurchaseDate = (event) => {
+		let date = null;
+		if (event.target.value === 'soon') {
+			date = 7;
+		} else if (event.target.value === 'kind-of-soon') {
+			date = 14;
+		} else if (event.target.value === 'not-soon') {
+			date = 30;
+		}
+		setDaysUntilNextPurchase(date);
+		setRadioSelect(event.target.value);
+	};
+
+	return (
+		<form onSubmit={onFormSubmit}>
+			<div>
+				<label htmlFor="item">Item Name:</label>
+			</div>
+			<input type="text" id="item" name="item" onChange={handleItemName} />
+			<div>How soon will you buy this again?</div>
+			<fieldset>
```

Comment:
So this is a pretty subtle accessibility thing, but you may want to consider doing one of two things to improve accessibility for this group of radio buttons.

The reason for this is that without either of them, assistive technology like screen readers would announce the labels on the individual radio buttons, but not necessarily the preceding div in context. I can explain this more in a weekly sync if y'all would like.

- One thing you could do is to change the `<div>How soon will you buy this again</div>` to a `<legend>` element inside the fieldset like this:
```html
<fieldset>
  <legend>How soon will you buy this again?</legend>
  <!-- ... -->
  <input type="radio"  />
  <!-- ... -->
</fieldset>
```
- Or, you could use the `aria-labelledby` attribute on each radio input referring to the div containing that label. See below:
```html
<div id="purchase-date-label">How soon will you buy this again?</div>
<fieldset>
  <!-- ... -->
  <input type="radio" aria-labelledby="purchase-date-label" />
  <!-- ... -->
</fieldset>
```
- See also: [WAI Tutorial on Grouping Form Controls](https://www.w3.org/WAI/tutorials/forms/grouping/#radio-buttons)

Reply:
We appreciate all of the feedback! Absolutely, we'd love to learn more about how to make the controls accessible in a weekly sync!

Reply:
@mtfoley Definitely new to the accessibility feedback.  Very important concerns I would like to be better about. Thank you!